worldedit-core/src/main/java/com/sk89q/worldedit/util/asset/AssetLoader.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.util.asset;
+
+import com.google.common.annotations.Beta;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.sk89q.worldedit.WorldEdit;
+import com.sk89q.worldedit.util.io.file.FilenameException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+@Beta
+public abstract class AssetLoader<T> {
+
+    private final Cache<String, T> assets = CacheBuilder.newBuilder()
+        .expireAfterAccess(5, TimeUnit.MINUTES)
+        .build();
+
+    private final WorldEdit worldEdit;
+    private final Path assetDir;
+
+    public AssetLoader(WorldEdit worldEdit, Path assetDir) {
+        this.worldEdit = worldEdit;
+        this.assetDir = assetDir;
+    }
+
+    /**
+     * Loads an asset.
+     *
+     * @param path path in assets directory, can be with and without its file extension
+     * @return asset if successfully loaded, null otherwise
+     */
+    @Nullable
+    public T getAsset(String path) {
+        // Remove file extension from mapped name
```

Comment:
So if I have `foo.jpg` and `foo.png`, they map to the same thing? Why isn't the full path used for caching?